src/Exceptionless.Web/Controllers/EventController.cs

```diff
@@ -892,26 +891,19 @@ ILoggerFactory loggerFactory
         if (filteredParameters == null || filteredParameters.Count == 0)
             return Ok();
 
-        if (projectId == null)
-            projectId = Request.GetDefaultProjectId();
+        projectId ??= Request.GetDefaultProjectId();
 
         // must have a project id
         if (String.IsNullOrEmpty(projectId))
-            return BadRequest("No project id specified and no default project was found.");
-
-        var project = Request.GetProject();
-        if (!String.Equals(project?.Id, projectId)) {
-            _logger.ProjectRouteDoesNotMatch(project?.Id, projectId);
-
-            project = await GetProjectAsync(projectId);
-
-            // Set the project for the configuration response filter.
-            Request.SetProject(project);
-        }
```

Comment:
Getting the project here doesn't make any sense because there is only three possible paths where the project gets set on the request (for use by the configuration header) and that's in this controller in 3 different actions...

This log message that was being outputted doesn't even make sense because it's outputted all the time when the project on the request is null. I think it was originally there to output if the request claims project id doesn't match the route project id.

Tokens can be scoped to a project id but there is no explicit tenancy check here on the project (only tenancy check to see if the user can access the associated projects organization).

Reply:
I'm all for removing the logger message. But it seems like we should try and get the already loaded project from the request if we can. Are you saying it's not possible for that to be populated in this endpoint?

Reply:
You are correct, it's never set before this point.